File: src/main/java/com/alexperal/maven/MavenProject.java
package com.alexperal.maven;

import java.util.ArrayList;
import java.util.List;

public class MavenProject {
    private List<Dependency> compileDeps = new ArrayList<>();
    private MavenProjectId id;

    public List<Dependency> compileDeps() {
        return compileDeps;
    }

    public void addRootDependency(Dependency dependency) {
        this.compileDeps.add(dependency);
    }

    public MavenProjectId getId() {
        return this.id;
    }
}

File: src/main/java/com/alexperal/maven/MavenProject.kt
package com.alexperal.maven

import com.google.common.graph.GraphBuilder
import com.google.common.graph.MutableGraph


class MavenProject {
    private val compileDeps: MutableGraph<Dependency> = GraphBuilder.directed().build()
    var id: MavenProjectId? = null

    fun dependencies(): MutableGraph<Dependency> {
        return compileDeps
    }

    fun addDependency(fatherDependency: Dependency?, dependency: Dependency) {
        this.compileDeps.addNode(dependency)
        fatherDependency?.let { this.compileDeps.putEdge(it, dependency) }
    }
}
